host/sgx/exception.c

```diff
@@ -46,8 +47,10 @@ uint64_t oe_host_handle_exception(oe_host_exception_context_t* context)
 
         oe_exception_record_t oe_exception_record = {0};
 
-        /* TODO PRP: We need to save the exception information in
-         * oe_exception_record */
+        // ATTN: the faulting address is expected to be read from the
```

Comment:
In this case, the simulated #FP will not be very useful. I can't think of any realistic exception handler of #PF that does not need to check the faulting address.

Reply:
Should we remove the simulation code instead?

Reply:
According to Peter, reporting faulty address at 0 is good enough for Java runtime. I think Java runtime depends on a SIGSEGV event, but not necessarily the address, to reach a safe point. In this regard, the simulated #FP event is still useful.

Reply:
Even for Java/.Net runtime code that handles #PF on NULL reference, reporting faulting address of 0 when the faulting address is non-zero might be problematic. For other code, such as stack overflow detection, reporting incorrect faulting address will render the code logic useless, or even cause unexpected result.